modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotConsensusFunctions.scala

```diff
@@ -78,6 +79,25 @@ object GlobalSnapshotConsensusFunctions {
         lastSignedArtifact.info.balances.getOrElse(address, Balance.empty).satisfiesCollateral(collateral)
       }
 
+    def isValidProposal(lastSignedArtifact: Signed[GlobalSnapshot], trigger: ConsensusTrigger)(
+      proposal: GlobalSnapshot
+    ): F[Boolean] = {
+      val events = proposal.blocks.unsorted.map(_.block.asRight[StateChannelOutput])
+
+      def proposalArtifact: F[GlobalSnapshot] = createProposalArtifact(lastSignedArtifact.ordinal, lastSignedArtifact, trigger, events)
+        .map(_._1)
+        .map { artifact =>
+          artifact.copy(
+            stateChannelSnapshots = proposal.stateChannelSnapshots,
+            info = artifact.info.copy(lastStateChannelSnapshotHashes = proposal.info.lastStateChannelSnapshotHashes)
+          )
```

Comment:
I think it wouldn't be a big deal if we would fix it right away instead doing it in two steps

Reply:
I'd like to keep those changes separated, as we don't have migration plan for testnet and I do not want to delay the deployment of this fix.

Reply:
Ok